### DISCUSSION TITLE
Accessing the register values

### DISCUSSION THREAD

**Discussion by fahadalisarwar1** [eModbus/eModbus#56] [eModbus applications]
I ran the modbus RTU example and it logs the modbus message response, I am interested in retrieving the actual register values from the modbus device. How do I convert the hex dump into values I can use in other parts of the program. Thanks

**Comment by bertmelis** [comment]
the hexadecimal values **are** the actual register values. you have to convert them yourself.

See this example: https://github.com/eModbus/eModbus/blob/master/examples/GenerateMessage/main.cpp

**Reply by fahadalisarwar1** [reply]
yes, but the dump shows the slave address and function code as well. 
![image](https://user-images.githubusercontent.com/32795677/105169155-6adc5000-5b1b-11eb-87b7-9df6cd831838.png)

I want to extract last 4 bytes. 
```
  uint16_t word1, word2;
  printf("%0x", response.get(4, word1));
  printf("%0x", response.get(6, word2));
```
Is this the correct way to extract them ?

**Reply by bertmelis** [reply]
What kind of return values are you expecting? 16-bits? 32-bits? signed/unsigned? floats?

**Reply by fahadalisarwar1** [reply]
32 bit unsigned which I can convert to IEE754 floating point.

**Reply by Miq1** [reply]
Why don't you use the float extraction function?
```
float f;
response.get(3, f);
```

**Reply by fahadalisarwar1** [reply]
yes thank you!
